File: java/src/test/java/jooqsimple/model/AbstractIntegrationTest.java
package jooqsimple.model;

import org.jooq.*;
import org.jooq.conf.*;
import org.jooq.impl.*;
import org.junit.jupiter.api.*;

import java.sql.*;

import static jooqsimple.DbConf.*;

abstract class AbstractIntegrationTest {

    private java.sql.Connection connection = null;

    DSLContext dslContext = null;

    @BeforeAll
    public void setup() throws SQLException {
        connection = DriverManager.getConnection(DBURL, DBUSERNAME, DBPASSWORD);
        dslContext = DSL.using(jooqConfiguration(connection));
    }

    private DefaultConfiguration jooqConfiguration(Connection connection) {
        var config = new DefaultConfiguration();
        config.set(SQLDialect.POSTGRES);
        config.set(connection);
        config.set(new PrettyPrinter());
        config.set(new Settings().withRenderFormatted(true).withRenderSchema(false));
        return config;
    }

    @AfterAll
    public void after() throws SQLException {
        if (connection != null) {
            connection.close();
        }
    }

}

File: kotlin/src/test/kotlin/jooqsimple/model/AbstractIntegrationTest.kt
package jooqsimple.model;

import jooqsimple.DBPASSWORD
import jooqsimple.DBURL
import jooqsimple.DBUSERNAME
import org.jooq.DSLContext
import org.jooq.SQLDialect
import org.jooq.conf.Settings
import org.jooq.impl.DSL
import org.jooq.impl.DefaultConfiguration
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import java.sql.Connection
import java.sql.DriverManager

abstract class AbstractIntegrationTest {

    lateinit var dslContext: DSLContext

    lateinit var connection: Connection

    @BeforeAll
    fun setup() {
        connection = DriverManager.getConnection(DBURL, DBUSERNAME, DBPASSWORD)
        dslContext = DSL.using(jooqConfiguration(connection))
    }

    private fun jooqConfiguration(connectionProvider: Connection?): DefaultConfiguration =
        DefaultConfiguration().apply {
            set(SQLDialect.POSTGRES)
            set(connectionProvider)
            set(PrettyPrinter())
            set(Settings().withRenderFormatted(true).withRenderSchema(false))
        }

    @AfterAll
    fun after() {
        connection.close()
    }

}
